### PR TITLE
Add hicolor-icon-theme and adwaita-icon-theme

### LINKED ISSUE #3918: Add hicolor-icon-theme
Would be good to add [`hicolor-icon-theme`]( https://www.freedesktop.org/wiki/Software/icon-theme/ ). This is required for some things in GTK 2 and 3.

### PR DESCRIPTION
**Pull request by ryanvolz**
These are needed so that applications using gtk3 have icons to use by default.

Closes #3918

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by ryanvolz** [comment]
Closes #3918

**Comment by ryanvolz** [comment]
@stuarteberg Consent to being a maintainer?

**Comment by stuarteberg** [comment]
I consent!

File: recipes/hicolor-icon-theme/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  build:
+    - make  # [unix]
+
+test:
+  commands:
+    - test -f $PREFIX/share/icons/hicolor/index.theme  # [unix]
+    - if not exist %PREFIX%\Library\share\icons\hicolor\index.theme exit 1  # [win]
+
+about:
+  home: hhttps://wiki.freedesktop.org/www/Software/icon-theme/
```
Comment (stuarteberg) [inline recipes/hicolor-icon-theme/meta.yaml:28]: typo here: `hhttps://`

**Comment by ryanvolz** [comment]
This does need a bit of re-work, as I've noticed now that this current recipe for adwaita-icon-theme does not render the svg icons into pngs (needs `gtk-encode-symbolic-svg` from `gtk3`). This maybe isn't strictly necessary because I think those pngs are used as a fallback. Other packagers solve the cyclic dependency by splitting `gtk3` into subpackages, which seems reasonable.

Additionally, it seems that for the svgs to be used at runtime, `librsvg` is needed as a run dep. That's currently fine for Linux/OSX, but it does mean making a Windows version of `librsvg`. Even if it's not used as a run dependency, `librsvg` is also needed for the build rendering so we'll need that Windows package either way.

**Comment by ryanvolz** [comment]
@conda-forge/staged-recipes This is ready for review.

File: recipes/adwaita-icon-theme/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  merge_build_host: False
+  ignore_run_exports:
+    - gtk3  # [win]
+    - librsvg  # [win]
+
+requirements:
+  build:
+    # gtk3 and librsvg here for rendering svg to png, build ok because platform independent
+    - gtk3  # [unix]
+    - librsvg  # [unix]
```
Comment (isuruf) [inline recipes/adwaita-icon-theme/meta.yaml:24]: Why are these needed in `build` ?

Comment (ryanvolz) [reply recipes/adwaita-icon-theme/meta.yaml:24]: They are needed for rendering some SVG icons to PNGs and for building the icon theme cache. I suppose they could go in host if that's what you're asking, but that function seemed more like a platform-independent build tool to me. Practically, we also don't want them as run dependencies, so if they are moved to host then ignore_run_exports will need to be set as with the Windows build.

**Comment by ryanvolz** [comment]
@conda-forge/staged-recipes Ping for review/merge.

**Comment by ryanvolz** [comment]
@conda-forge/staged-recipes Ping for review/merge.

File: recipes/adwaita-icon-theme/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    - make  # [unix]
+    - python
+  host:
+    # windows build files only know of one prefix, so these need to be in host even
+    # though we don't link against them and they are functionally build tools
```
Comment (isuruf) [inline recipes/adwaita-icon-theme/meta.yaml:29]: I'd suggest removing these and adding `merge_build_host: True # [win]`

Comment (ryanvolz) [reply recipes/adwaita-icon-theme/meta.yaml]: Good idea!

File: recipes/adwaita-icon-theme/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - gtk3  # [win]
+    - librsvg  # [win]
+
+requirements:
+  build:
+    # gtk3 and librsvg here for rendering svg to png, build ok because platform independent
+    - gtk3
+    - librsvg
+    - make  # [unix]
+    - python
```
Comment (isuruf) [inline recipes/adwaita-icon-theme/meta.yaml:28]: Can you add an empty `host` here?

Comment (ryanvolz) [reply recipes/adwaita-icon-theme/meta.yaml:28]: The subsequent commit works, right?

**Comment by ryanvolz** [comment]
@isuruf Build succeeded. Anything else I should do?

**Comment by ryanvolz** [comment]
@isuruf Thanks!
